lib/active_merchant/billing/gateways/pay_trace.rb

```diff
@@ -0,0 +1,259 @@
+          add_customer_data(post, options)
+
+          response = commit('transactions/authorization/keyed', post)
+          check_token_response(response, 'transactions/authorization/keyed', post, options)
+        end
+      end
+
+      def capture(authorization, options = {})
+        post = {}
+        post[:amount] = options[:amount] if options[:amount]
+        post[:transaction_id] = authorization
+        response = commit('transactions/authorization/capture', post)
+        check_token_response(response, 'transactions/authorization/capture', post, options)
+      end
+
+      def refund(authorization, options = {})
+        # currently only support full and partial refunds of settled transactions via a transaction ID
+        post = {}
+        post[:amount] = options[:amount] if options[:amount]
+        post[:transaction_id] = authorization
+        response = commit('transactions/refund/for_transaction', post)
+        check_token_response(response, 'transactions/refund/for_transaction', post, options)
+      end
+
+      def void(authorization, options = {})
+        post = {}
+        post[:transaction_id] = authorization
+
+        response = commit('transactions/void', post)
+        check_token_response(response, 'transactions/void', post, options)
+      end
+
+      def verify(credit_card, options = {})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      # The customer_IDs that come from storing cards can be used for auth and purchase transaction types
+      def store(credit_card, options = {})
+        post = {}
+        post[:customer_id] = options[:customer_id] || SecureRandom.hex(12)
+        add_payment(post, credit_card)
+        add_address(post, credit_card, options)
+        response = commit('customer/create', post)
+        check_token_response(response, 'customer/create', post, options)
+      end
+
+      def redact(customer_id)
+        post = {}
+        post[:customer_id] = customer_id
+        response = commit('customer/delete', post)
+        check_token_response(response, 'customer/delete', post, options)
+      end
+
+      def settle
+        post = {}
+        response = commit('transactions/settle', post)
+        check_token_response(response, 'transactions/settle', post, options)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Bearer )[a-zA-Z0-9:_]+), '\1[FILTERED]').
+          gsub(%r(("credit_card\\?":{\\?"number\\?":\\?")\d+), '\1[FILTERED]').
+          gsub(%r(("cvv\\?":\\?")\d+), '\1[FILTERED]').
+          gsub(%r(("username\\?":\\?")\w+@+\w+.+\w+), '\1[FILTERED]').
+          gsub(%r(("password\\?":\\?")\w+), '\1[FILTERED]').
+          gsub(%r(("integrator_id\\?":\\?")\w+), '\1[FILTERED]')
+      end
+
```

Comment:
I think we could move this into `private`; it would be neat if we could figure out a way to test `acquire_access_token` without it having to be public 🤔

Reply:
Moved to `private`! Will chat you up tomorrow about the latter part of this comment

Reply:
Also together!